components/files/file.jsx

```diff
@@ -0,0 +1,152 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Files design pattern](https://lightningdesignsystem.com/components/files/) in React.
+// Based on SLDS v2.4.0
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+import { FILES_FILE } from '../../utilities/constants';
+
+import FileFigure from './private/file-figure';
+import FileActions from './private/file-actions';
+
+const displayName = FILES_FILE;
+
+const propTypes = {
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	crop: PropTypes.oneOf(['16-by-9', '4-by-3', '1-by-1']),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 *  Action to be done on clicking download button; doesnt show download button if empty
+	 */
+	onClickDownload: PropTypes.func,
+	/**
+	 *  Action to be done on clicking more actions button; doesn't show More actions button if empty
+	 */
+	onClickMoreActions: PropTypes.func,
+	/**
+	 *  Link to External Icon
+	 */
+	externalIcon: PropTypes.node,
+	/**
+	 *  Link to File Thumbnail
+	 */
+	image: PropTypes.string,
+	/**
+	 *  Whether file preview is loading
+	 */
+	isLoading: PropTypes.bool,
+	/**
+	 *  Action to be taken on clicking on image
+	 */
+	href: PropTypes.string,
+	/**
+	 *  Title for the File
+	 */
+	title: PropTypes.string,
+};
+
+const defaultProps = {
+	isLoading: false,
+	crop: '16-by-9',
```

Comment:
Add a default value for `href` here of `javascript:void(0);`, that way you can remove setting it in every example.